linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -115,6 +115,22 @@ void Process ()
 				ProcessQueue ();
 				ProcessVirtualMethods ();
 			}
+
+			// deal with [TypeForwardedTo] pseudo-attributes
+			foreach (AssemblyDefinition assembly in _context.GetAssemblies ()) {
+				if (!assembly.MainModule.HasExportedTypes)
+					continue;
+
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					if (!exported.IsForwarder)
+						continue;
+					var type = exported.Resolve ();
+					if (!Annotations.IsMarked (type))
+						continue;
+					Annotations.Mark (exported);
+					Annotations.Mark (assembly.MainModule);
```

Comment:
Right, things can be a bit easier for XI since we can assume there will be no code generation (or download) so the linker sees everything. That is not a correct assumption outside XI fullAOT compilation. That piece of code might be better left for XI (not mobile) alone - and not used in .net core.

There's also an issue in C9 (Cecil 0.10) where some Scope are not updated (on exported types). That did not happen with C8 (Cecil 0.9) but it might not be related to Cecil (I'm investigating this).